src/main/java/com/zufar/onlinestore/payment/service/PaymentCreator.java

```diff
@@ -0,0 +1,48 @@
+package com.zufar.onlinestore.payment.service;
+
+import com.stripe.exception.StripeException;
+import com.stripe.model.PaymentIntent;
+import com.zufar.onlinestore.payment.dto.PaymentDetailsWithTokenDto;
+import com.zufar.onlinestore.payment.entity.Payment;
+import com.zufar.onlinestore.payment.enums.PaymentConstants;
+import com.zufar.onlinestore.payment.mapper.PaymentConverter;
+import com.zufar.onlinestore.payment.repository.PaymentRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class PaymentCreator {
+
+    private final PaymentIntentCreator paymentIntentCreator;
+    private final PaymentRepository paymentRepository;
+    private final PaymentConverter paymentConverter;
+
+    public PaymentDetailsWithTokenDto createPayment(String paymentMethodId, BigDecimal totalPrice, String currency) throws StripeException {
+        PaymentIntent paymentIntent = paymentIntentCreator.createPaymentIntent(paymentMethodId, totalPrice, currency);
+        log.info("create payment: payment intent successfully created: paymentIntentId: {}.", paymentIntent.getId());
+        String paymentToken = paymentIntent.getClientSecret();
+        Payment payment = fillPaymentDetails(paymentIntent);
+        Payment savedPayment = paymentRepository.save(payment);
+        log.info("create payment: payment successfully saved: savedPayment: {}.", savedPayment);
+
+        return PaymentDetailsWithTokenDto.builder()
+                .paymentToken(paymentToken)
+                .paymentDetailsDto((paymentConverter.toDto(savedPayment)))
+                .build();
+    }
+
+
+    private Payment fillPaymentDetails(PaymentIntent paymentIntent) {
```

Comment:
I think it is task of mapper or converter which you can extract as a separate class like `PaymentConverter` with method `public Payment toPayment(PaymentIntent paymentIntent)`.

Reply:
done